Add an integration test for LASR using pipe and chunked decoding

kho opened the pull request:
# What does this PR do?

- Update the goldens for `test_model_integration_batched`
- Add an integration test for LASR using pipe and chunked decoding.

This is separated from https://github.com/huggingface/transformers/pull/42720 per request from @eustlb.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

kho committed:
Add an integration test for LASR using pipe and chunked decoding

eustlb commented:
run-slow: lasr

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_42823). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.

kho committed:
Revise goldens in LasrForCTCIntegrationTest.test_model_integration_batched

kho committed:
Enable LasrForCTCIntegrationTest

eustlb reviewed (approved):
LGTM, thanks! running slow tests to ensure values are the correct ones on CI devices.

eustlb committed:
Merge branch 'main' into pipe_with_chunk_test

eustlb commented:
run-slow: lasr

eustlb committed:
Merge branch 'main' into pipe_with_chunk_test

eustlb committed:
Merge branch 'main' into pipe_with_chunk_test

eustlb committed:
add require_torch_accelerator

eustlb committed:
Merge branch 'main' into pipe_with_chunk_test

eustlb commented:
run-slow: lasr

kho committed:
Use a publicly accessible test model for LASR and update integration test goldens

kho commented on tests/models/lasr/test_modeling_lasr.py:
```diff
@@ -295,12 +302,13 @@ def test_sdpa_can_dispatch_composite_models(self):
     @classmethod
     def setUp(cls):
-        cls.checkpoint_name = "eustlb/lasr"
+        cls.checkpoint_name = "hf-internal-testing/lasr-test"
```
@eustlb The test checkpoint is https://huggingface.co/wuketest/lasr_integration_test. Could you help me make a copy to `hf-internal-testing/lasr-test`?

eustlb reviewed (approved):
thanks @kho! running slow tests before merging to see whether we need to update value for the runners

eustlb commented:
run-slow: lasr

eustlb committed:
Merge branch 'main' into pipe_with_chunk_test

eustlb committed:
Merge branch 'main' into pipe_with_chunk_test

kho committed:
Merge branch 'main' into pipe_with_chunk_test

kho committed:
Correct the tokenizer mapping for LASR models

Because of the out of date tokenizer mapping, AutoTokenizer started returning TokenizersBackend instead LasrTokenizer after https://github.com/huggingface/transformers/pull/42894, which caused https://github.com/Google-Health/medasr/issues/12.

kho committed:
Remove require_read_token since we now use a publicly assessible test checkpoint

eustlb committed:
update values for runners

eustlb reviewed (approved):
Thanks a lot @kho! I've udpated values for our runners
